### ISSUE TITLE
Table of contents section has role=dialog on desktop

### ISSUE THREAD

**Issue by tombye** [alphagov/tech-docs-gem#357]
## What should change

The table of contents shouldn't have role=dialog when it doesn't have dialog behaviours (on screens larger than mobile size).

## User need

On mobile, the table of contents section acts like a [dialog](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role) because It:
- is hidden by default
- is shown when you activate a button
- is focused when shown
- has an accessible name
- constrains tabbing within the dialog

It has a role=dialog, which makes sense there.

<img width="318" alt="image" src="https://github.com/user-attachments/assets/78fb2980-3a55-4293-8deb-6e942044d198">

On desktop, it doesn't behave like this so shouldn't have that role:

<img width="1727" alt="image" src="https://github.com/user-attachments/assets/67b7e4fd-8997-438b-b7cf-76599c944713">

This effectively leaves it with the wrong semantics which, I think, would make it fail [success criterian 1.3.1 - info and relationships](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html).

**Comment by tombye** [comment]
This has been fixed by https://github.com/alphagov/tech-docs-gem/pull/358 so closing.
